### PR TITLE
Introduces prettier and changes from standard cli to eslint

### PR DESCRIPTION
**Pull request by mikaelbr**
This PR introduces prettier with same settings as cerebral main repo. To achieve this in a consistent way it changes from using the standard convenience cli to use eslint directly.

Also adds eslint fixing on git hooks and prettier on non-javascript-files.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      "error",
+      {
+        "trailingComma": "es5",
+        "singleQuote": true,
+        "semi": false
+      }
+    ],
+    "standard/computed-property-even-spacing": "off"
+  }
+}
```
Comment (henri-hulski) [inline .eslintrc.json]: Missing newline at EOF. :)

**Review by henri-hulski — APPROVED** [review]
Great work!
Just fix the missing empty line at EOF.
Maybe we should also consider to add css and md linting to Cerebral.

**Comment by mikaelbr** [comment]
Did some updating, and IMO improvements in the eslint configuration. Extracted the prettier settings from eslint config to be placed in prettierrc so it can be used in editor settings and such. Also added a `.vscode` editor setting files with formatOnSave true for enforcing styles actively while coding.

> Maybe we should also consider to add css and md linting to Cerebral.

Note, this isn't introducing linting to CSS and md files, just pretty printing (parsing + re-print). In some way it validates the code, as it parses ASTs, and often this is enough. But it isn't as strict as linting. For that you could do something like stylelint for CSS.

**Comment by Guria** [comment]
I don't like editor specific files in shared repo. Don't all use same editors. If we adding vscode config, does it mean we should place webstorm, atom, vim, emacs configs as well? 
We still can have a section in readme with recommended settings.

**Comment by mikaelbr** [comment]
Sure. There are trade-offs, different opinions on the subject. Dropped commit.

**Comment by mikaelbr** [comment]
>  Editors should be able to use .eslintrc.json directly. Just use the eslint plugin for your editor and at least for javascript it works out of the box.

As you say this is oriented around linting. Linting tools for editors can read eslint configuration inherited by prettier config, but again pretty printing isn't the same as eslint. 

If you use it through eslint configuration the configurations wouldn't be read for formatting things as CSS files and markdown files. For this project that would mean that 16% of the code base would either not be pretty printed, or pretty printed with different settings.

I would argue that this is a more idiomatic way of using prettier, and that the Cerebral project should change it's setup if the point is to have it the same. Ideally, I'd say that using linting for style rules when having AST pretty printing is unnecessary and a overuse of linting.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "ecmaFeatures": {
+      "jsx": true
+    }
+  },
+  "env": {
+    "browser": true,
+    "node": true,
+    "es6": true
+  },
+  "plugins": ["prettier"],
```
Comment (henri-hulski) [inline .eslintrc.json]: When using recommended,  `"plugins": ["prettier"],` can be removed as it's already [part of recommended](https://github.com/prettier/eslint-plugin-prettier/blob/master/README.md#recommended-configuration).

**Review by henri-hulski — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 2 above)
Comment (mikaelbr) [reply .eslintrc.json]: Oh, you're right. Missed that 👍

**Comment by henri-hulski** [comment]
Yeah you're right. It seems the recommendations changed since I set up Cerebral.
I think the `recommended` option is new.

**Comment by mikaelbr** [comment]
What one could do is to extract the eslint config to a seperate repo/module and reuse (extend through eslint) on Cerebral and other projects. Easier to maintain and be consistant. Though cerebral I guess is a mono-repo, there are a lot of other repos as well. AFAIK, you'd have to duplicate the prettier rc-file, though.

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "trailingComma": "es5",
+  "singleQuote": true,
+  "semi": false,
+  "printWidth": 80
```
Comment (henri-hulski) [inline .prettierrc:5]: Hmm seems that I didn't saved my comment here.
I think we can leave out `"printWidth": 80` as it's the default.

**Comment by christianalfoni** [comment]
@mikaelbr You have time for a rebase here? :) Will release new version of debugger ASAP

**Comment by mikaelbr** [comment]
I’ll try to find the time within the end of the day. If it’s just squashing you are thinking of, you can also do that through the GH ui.

**Review by henri-hulski — APPROVED** [review]

**Comment by henri-hulski** [comment]
@mikaelbr  BTW thanks for inspiration.
cerebral/cerebral#1295

**Comment by mikaelbr** [comment]
@christianalfoni & @henri-hulski : How do you want this in regards to commits? Should I squash something, reorder, reword?

**Comment by mikaelbr** [comment]
Btw, @christianalfoni , I think I got all merge conflicts, but would love it if you did a quick check and see that your latest changes is still good.

**Comment by henri-hulski** [comment]
For me the commits are OK. Other than in Cerebral here we don't use the commit messages to create the release notes.

**Comment by christianalfoni** [comment]
Great, I will do a small check, thanks a bunch @mikaelbr :-)

**Pull request merged**